tests/migrations/test_autodetector.py

```diff
@@ -2257,3 +2257,49 @@ def test_add_non_blank_textfield_and_charfield(self, mocked_ask_method):
         self.assertNumberMigrations(changes, 'testapp', 1)
         self.assertOperationTypes(changes, 'testapp', 0, ["AddField", "AddField"])
         self.assertOperationAttributes(changes, 'testapp', 0, 0)
+
+    def test_unmanaged_model_fk_to_managed_model(self):
+        managed = ModelState("testapp", "ManagedModel", [
+            ("id", models.AutoField(primary_key=True)),
+        ])
+        unmanaged_with_fk = ModelState("testapp", "UnmanagedModel", [
+            ("id", models.AutoField(primary_key=True)),
+            ("name", models.ForeignKey("testapp.ManagedModel", models.CASCADE)),
+        ], {'managed': False})
+        changes = self.get_changes([], [managed, unmanaged_with_fk])
+        self.assertNumberMigrations(changes, 'testapp', 1)
+        self.assertOperationTypes(changes, 'testapp', 0, ["CreateModel", "CreateModel", "AddField"])
+        # I have no idea how to get assertOperationFieldAttributes to play well
+        # with verifying what name *points to*, so this'll have to do.
+        self.assertOperationAttributes(changes, 'testapp', 0, 2, name="name")
+
+    def test_unmanaged_model_fk_to_another_managed_model(self):
+        unmanaged = ModelState("testapp", "RelatedUnmanagedModel", [
+            ("id", models.AutoField(primary_key=True)),
+        ], {'managed': False})
+        unmanaged_with_fk = ModelState("testapp", "UnmanagedModel", [
+            ("id", models.AutoField(primary_key=True)),
+            ("name", models.ForeignKey("testapp.RelatedUnmanagedModel", models.CASCADE)),
+        ], {'managed': False})
+        changes = self.get_changes([], [unmanaged, unmanaged_with_fk])
+        self.assertNumberMigrations(changes, 'testapp', 1)
+        self.assertOperationTypes(changes, 'testapp', 0, ["CreateModel", "CreateModel"])
+        # I have no idea how to get any of the asserts* methods to pluck
+        # out the list of fields on a CreateModel
+        fields_on_unmanaged_with_fk = {_[0] for _ in changes['testapp'][0].operations[1].fields}
+        self.assertEqual(fields_on_unmanaged_with_fk, {'id', 'name'})
+
+    def test_adding_an_fk_to_an_unmanaged_model(self):
```

Comment:
Can you add a similar test case where the FK points to a managed model, please :)